pkg/process/metadata/parser/java/yaml.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
```

Comment:
have you tried to use 3rd party package for this logic?
https://pkg.go.dev/github.com/jeremywohl/flatten for example

Reply:
I could use that package to flatten the map key. The issue that is remaining is more likely parsing from the yaml. So I preferred to flatten myself while parsing and reduce the number of entries I have since I can skip for instance arrays that I won't need (e.g. spring application name is never on an array, it's a scalar property).
Generally speaking one optimization could be not to hold all the values on the map but only search for `spring.application.name`. This will anyway exclude resolving that value for applications that are using placeholder substitutions. An example:
`spring.application.name=${my.cool.property)` At this point I also need the value for `my.cool.property` to fully resolve the chain

Reply:
then just like https://github.com/DataDog/datadog-agent/pull/22768#discussion_r1505312535, I think that should be reflected in the documentation in file

Reply:
It's a common thing for spring users but I can understand it should better documented. I put some comments [here](https://github.com/DataDog/datadog-agent/blob/82bb63b7b4d333b911393424c02f9e2e636cc32a/pkg/process/metadata/parser/java/spring.go#L137) and [here](https://github.com/DataDog/datadog-agent/blob/82bb63b7b4d333b911393424c02f9e2e636cc32a/pkg/process/metadata/parser/java/spring.go#L158)